### ISSUE TITLE
Interop: cleanup op-node interop setup

### ISSUE THREAD

**Issue by protolambda** [ethereum-optimism/optimism#13338]
After implementing ManagedMode and StandardMode, we can drop the temporary work-around mode.

Blocked by:
- #13336 
- #13337 

`op-node/rollup/interop/config.go`:
- update `Check` to only allow one mode of operation at a time
- remove `TemporarySetup` (blocked by implementing Managed & Standard mode)

`op-node/rollup/interop/tmpapi.go`:
- remove `TemporaryInteropServer`
- remove `TemporaryInteropAPI`

**Comment by protolambda** [comment]
Work in progress @ https://github.com/ethereum-optimism/optimism/pull/13406

**Comment by axelKingsley** [comment]
I believe this is invalidated by the Standard Mode tasks, plus the fact that this task hasn't been meaningfully updated since early December. @protolambda is there still some valid information in here or shall we close this?

**Comment by protolambda** [comment]
Let's close it. I checked the work in the ticket and believe it's already done. The standard-mode work is tracked separately.
